### DISCUSSION TITLE
Support for pnpm

### DISCUSSION THREAD

**Discussion by ttu-ttu** [cyrilletuzi/vscode-angular-schematics#735] [Ideas]
Would be nice if the extension supports Angular projects that use `pnpm`, it currently fails to detect `@schematics/angular` as it is located under `node_modules/.pnpm` (plus something else)

Current fix is to manually include `@schematics/angular` to `package.json`, delete `node_modules` and reinstall packages, just to make it appear in `node_modules` directly

**Comment by cyrilletuzi** [comment]
Hello, support for pnpm is planned in an extended version coming soon. I'll post more details when it becomes available.

**Comment by cyrilletuzi** [comment]
Hi @ttu-ttu, for information, pnpm support is now available in **Schematics Pro**, an extended version of the extension. You will find all the details on the [dedicated website](https://www.cyrilletuzi.com/schematics-pro/).

To be exact, Schematics Pro totally avoids the issue by including its own schematics generator, so it is not dependent at all on `node_modules`. It also has the benefit to be faster and to avoid other issues (like shell issues, and not requiring a global Angular CLI).

**Reply by ttu-ttu** [reply]
I see, thanks for letting me know!

**Comment by cyrilletuzi** [comment]
Latest version (v6) of the extension does not rely on `node_modules` anymore, and so it compatible with alternate package managers.
